integration_tests/pages/barcode/createNewContact.ts

```diff
@@ -14,17 +14,55 @@ export default class CreateNewContactPage extends Page {
   }
 
   submitWithValidValues = (): ReviewRecipientsPage => {
+    this.enterAValidPrisonerName()
+    this.typeAheadAValidPrison()
+    return this.submitForm(ReviewRecipientsPage)
+  }
+
+  enterAValidPrisonerName = (): CreateNewContactPage => {
+    this.prisonerNameField().clear()
     this.prisonerNameField().type('Gage Hewitt')
+    return this
+  }
+
+  typeAheadAValidPrison = (): CreateNewContactPage => {
+    this.clearPrisonField()
     this.prisonIdField().type('ashfield')
     this.pressEnterInPrisonIdField()
-    this.submitButton().click()
-    return Page.verifyOnPage(ReviewRecipientsPage)
+    return this
+  }
+
+  typeAheadAnInvalidPrison = (): CreateNewContactPage => {
+    this.clearPrisonField()
+    this.prisonIdField().type('not a valid prison')
+    this.pressEnterInPrisonIdField()
+    return this
+  }
+
+  clearPrisonField = (): CreateNewContactPage => {
+    this.prisonIdField().clear()
+    return this
   }
 
   pressEnterInPrisonIdField = () => {
     this.prisonIdField().type('\n')
   }
 
+  submitForm<T>(T): T {
```

Comment:
I _think_ this is how you do generics in typescript. It means we can have one `submit` method, and you call it with the class name you expect to land on - eg: `submitForm(FindRecipientsPage)` or `submitForm(CreateContactPage)` etc.

As a separate improvement maybe we consider moving this `submitForm` method to the superclass (`Page`) and adding an abstract `submitButton` method to the `Page` class which all subclasses need to implement. That might be quite nice?

Reply:
Agreed, maybe in another PR try it on the Page clas. Very nice!